src/Microsoft.TestPlatform.CrossPlatEngine/Client/Parallel/ParallelRunEventsHandler.cs

```diff
@@ -54,12 +54,32 @@ internal class ParallelRunEventsHandler : ITestRunEventsHandler
         /// <summary>
         /// Handles the Run Complete event from a parallel proxy manager
         /// </summary>
-        public void HandleTestRunComplete(
+        public virtual void HandleTestRunComplete(
             TestRunCompleteEventArgs testRunCompleteArgs,
             TestRunChangedEventArgs lastChunkArgs,
             ICollection<AttachmentSet> runContextAttachments,
             ICollection<string> executorUris)
         {
+            var parallelRunComplete = HandleSingleTestRunComplete(testRunCompleteArgs, lastChunkArgs, runContextAttachments, executorUris);
+
+            if (parallelRunComplete)
+            {
+                var completedArgs = new TestRunCompleteEventArgs(runDataAggregator.GetAggregatedRunStats(),
+                    runDataAggregator.IsCanceled,
+                    runDataAggregator.IsAborted,
+                    runDataAggregator.GetAggregatedException(),
+                    new Collection<AttachmentSet>(runDataAggregator.RunCompleteArgsAttachments),
```

Comment:
Is it required to create a new Collection here, can we not just pass runContextAttachments?
`runContextAttachments as Collection<AttachmentSet>`